crates/types/src/traits/election.rs

```diff
@@ -23,33 +23,45 @@ pub trait Membership<TYPES: NodeType>:
         committee_topic: Topic,
     ) -> Self;
 
-    /// Get all participants in the committee (including their stake)
-    fn stake_table(&self) -> Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>;
+    /// Get all participants in the committee (including their stake) for a specific epoch
+    fn stake_table(
+        &self,
+        epoch: TYPES::EpochTime,
```

Comment:
I guess all of these functions will now have to return a `Result`, because we may not have the stake table for a given (future) epoch?

(since this was quite an involved change, I can do it in #3738 after merging this -- just asking for clarification)